src/btree/bt_page.c

```diff
@@ -62,6 +62,13 @@ __wt_page_alloc(
 
     WT_RET(__wt_calloc(session, 1, size, &page));
 
+    /* Initialize the spinlock for the page. */
+    ret = __wt_spin_init(session, &page->page_lock, "btree page");
```

Comment:
Error handling in this function is now spread across two locations, but the existing error handling being inside a switch statement was already unusual. I'll spin up a neweng ticket to clean this up outside of the PR.